pkg/controllers/provisioningv2/fleetcluster/fleetcluster_test.go

```diff
@@ -0,0 +1,236 @@
+package fleetcluster
+
+import (
+	"fmt"
+	"testing"
+
+	fleet "github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	mgmt "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	mgmtv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	errNotImplemented = fmt.Errorf("unimplemented")
+	errNotFound       = fmt.Errorf("not found")
+
+	builtinAffinity = corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+				{
+					Weight: 1,
+					Preference: corev1.NodeSelectorTerm{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "fleet.cattle.io/agent",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	linuxAffinity = corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "kubernetes.io/os",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"linux"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	resourceReq = &corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU: resource.MustParse("1"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("1Gi"),
+		},
+	}
+	tolerations = []corev1.Toleration{
+		{
+			Key:      "key",
+			Operator: corev1.TolerationOpEqual,
+			Value:    "value",
+		},
+	}
+)
+
+func TestCluster(t *testing.T) {
+	require := require.New(t)
+
+	h := &handler{
+		clustersCache:     &fakeClusterCache{},
+		getPrivateRepoURL: func(*v1.Cluster, *mgmt.Cluster) string { return "" },
+	}
+
+	cluster := &v1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster", Namespace: "test-namespace",
+		},
+		Spec: v1.ClusterSpec{},
+	}
+	clusterStatus := v1.ClusterStatus{ClusterName: "cluster-name", ClientSecretName: "client-secret-name"}
+
+	labels := map[string]string{"cluster-group": "cluster-group-name"}
+
+	tests := []struct {
+		name          string
+		cluster       *v1.Cluster
+		status        v1.ClusterStatus
+		clustersCache mgmtv3.ClusterCache
+		wantLen       int
```

Comment:
Just preference, but I usually use `expected` like `expectedLen` and `expectedFleet` because it follows the actual/expected structure.

Reply:
I can change this, but I picked that name because  a lot of tests in rancher/rancher use this style.
I can see more tests are using expected (~359) vs want (~250).